source/Fleet.cpp

```diff
@@ -203,25 +204,57 @@ void Fleet::Load(const DataNode &node)
 			{
 				resetVariants = false;
 				variants.clear();
+				stockVariants.clear();
 				total = 0;
+				stockTotal = 0;
 			}
-			variants.emplace_back(child);
-			total += variants.back().weight;
+
+			int weight = 1;
+			string variantName;
+			bool named = false;
+			if(child.Size() >= 2 + add && !child.IsNumber(1 + add))
+			{
+				variantName = child.Token(1 + add);
+				named = true;
+			}
+			if(child.Size() >= 2 + add + named)
+				weight = child.Value(1 + add + named);
+
+			total += weight;
+			if(named)
+			{
+				stockVariants.emplace_back(GameData::Variants().Get(variantName), weight);
+				stockTotal += weight;
+			}
+			else
+				variants.emplace_back(child, weight);
 		}
 		else if(key == "variant")
 		{
-			// If given a full ship definition of one of this fleet's variant members, remove the variant.
+			// If given a full definition of one of this fleet's variant members, remove the variant.
 			bool didRemove = false;
 			Variant toRemove(child);
-			for(auto it = variants.begin(); it != variants.end(); ++it)
-				if(toRemove.ships.size() == it->ships.size() &&
-					is_permutation(it->ships.begin(), it->ships.end(), toRemove.ships.begin()))
-				{
-					total -= it->weight;
-					variants.erase(it);
-					didRemove = true;
-					break;
-				}
+			// If toRemove is named, check if a stockVariant shares that name.
+			// Else, check if toRemove is equal to any of the variants.
```

Comment:
Shouldn't the remove remove both the stockvariants and the variant?

Reply:
What do you mean by that? `remove variant` can only match either a stockVariant or a normal variant. There won't be both a stockVariant and normal variant that match.